sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -319,7 +382,8 @@ impl Plan {
             services.insert(addr, allocation);
         }
 
-        let plan = Self { services };
+        let dns_config = dns_builder.build();
```

Comment:
This new format looks good to me - as mentioned above, should we be adding the sled underlay addresses themselves to these initial DNS records?